File: src/game/level/world1/World1Level5.java
package game.level.world1;

import game.Bomberman;
import game.entity.Player;
import game.entity.enemies.boss.Boss;
import game.entity.enemies.boss.clown.Clown;
import game.entity.enemies.boss.ghost.GhostBoss;
import game.entity.enemies.npcs.FlyingEnemy;
import game.entity.enemies.npcs.Helicopter;
import game.entity.enemies.npcs.YellowBall;
import game.entity.models.Coordinates;
import game.entity.models.Enemy;
import game.level.Level;
import game.level.WorldSelectorLevel;
import game.level.world2.World2Level;
import game.level.world2.World2Level1;
import game.ui.panels.game.PitchPanel;
import game.utils.Paths;

import java.awt.*;

import static game.ui.panels.game.PitchPanel.GRID_SIZE;

public class World1Level5 extends World1Level{
    @Override
    public int getLevelId() {
        return 5;
    }

    @Override
    public Boss getBoss() {
        return new GhostBoss();
    }

    @Override
    public int startEnemiesCount() {
        return 0;
    }

    @Override
    public Class<? extends Enemy>[] availableEnemies() {
        return new Class[]{
                YellowBall.class,
                Helicopter.class
        };
    }

    @Override
    public boolean isLastLevelOfWorld(){
        return true;
    }

    @Override
    public Class<? extends Level> getNextLevel() {
        return World2Level1.class;
    }

    @Override
    public Coordinates getPlayerSpawnCoordinates() {
        return Coordinates.fromRowAndColumnsToCoordinates(new Dimension(0, (int)PitchPanel.DIMENSION.getHeight()/GRID_SIZE-1));
    }
}

File: src/game/level/world1/World1Level5.kt
package game.level.world1

import game.entity.enemies.boss.Boss
import game.entity.enemies.boss.ghost.GhostBoss
import game.entity.enemies.npcs.Helicopter
import game.entity.enemies.npcs.YellowBall
import game.entity.enemies.npcs.Zombie
import game.entity.models.Coordinates
import game.entity.models.Enemy
import game.level.Level
import game.level.StoryLevel
import game.level.info.model.LevelInfo
import game.level.info.imp.World1LevelInfo
import game.level.world2.World2Level1
import game.ui.panels.game.PitchPanel
import java.awt.Dimension

class World1Level5 : StoryLevel() {
    override val info: LevelInfo
        get() = object : World1LevelInfo(this) {
            override val levelId: Int get() = 5
            override val boss: Boss get() = GhostBoss()
            override val startEnemiesCount: Int get() = 0
            override val availableEnemies: Array<Class<out Enemy>> get() = arrayOf(YellowBall::class.java, Helicopter::class.java, Zombie::class.java)
            override val isLastLevelOfWorld: Boolean get() = true
            override val nextLevel: Class<out Level?> get() = World2Level1::class.java
            override val playerSpawnCoordinates: Coordinates
                get() = Coordinates.fromRowAndColumnsToCoordinates(
                            Dimension(0,
                                    PitchPanel.DIMENSION.getHeight().toInt() / PitchPanel.GRID_SIZE - 1)
                    )
        }
}